### ISSUE TITLE
remote start-tunnel error when connect device with usbfluxd

### ISSUE THREAD

**Issue by li8911255** [doronz88/pymobiledevice3#930]
**Test environment**
- macOS  Ventura 13.5
- iPhone 13 Pro , iOS 17.2.1

**Describe the bug**
After connect device from usbfluxd,  remote start-tunnel return 

ERROR Device is not connected

but usbmux list can print my device info correctly.

**Comment by doronz88** [comment]
Please provide exact steps to reproduce

**Comment by li8911255** [comment]
@doronz88 
> Please provide exact steps to reproduce

**To Reproduce**
Steps to reproduce the behavior:
1. Two macs located on the same LAN
2. Mac 1:Launch usbfluxd with root privileges (sudo ./usbfluxd -f),then launch socat to transferring data between addresses(socat tcp-listen:5000,fork unix-connect:/var/run/usbmuxd)
3. Mac 2:Connect to the remote Mac 1 (sudo ./usbfluxd -f -r #Mac 1 ip:5000)
4. now Mac 2 can list Mac 1 physically connected iOS device(Both idevice_id -l and pymobiledevice3 usbmux list),but pymobiledevice3 remote start-tunnel return ERROR Device is not connected

**Comment by doronz88** [comment]
I don't have two macs to test this with.
However, if you're working with corellium you don't need usbflux at all. You can simply pass `--usbmux` option.

For example:
```shell
pymobiledevice3 syslog live --usbmux IP:5000
```

Also, if you need tunnel establishment, the easier way would be to use lockdown tunnels but that depends on iOS >= 17.4.

```shell
sudo pymobiledevice3 lockdown start-tunnel --usbmux IP:5000
```

This should also prove much faster since it doesn't go through any middleware.

I had problems implementing the remote pairing required for their devices since they did something weird to the device pairing process.

**Comment by kelvin321** [comment]
> **Test environment**
> 
>     * macOS  Ventura 13.5
> 
>     * iPhone 13 Pro , iOS 17.2.1
> 
> 
> **Describe the bug** After connect device from usbfluxd, remote start-tunnel return
> 
> ERROR Device is not connected
> 
> but usbmux list can print my device info correctly.

remote tunnel requires IPv6 address (and a bonjour service on this address) provided by NCM, your iPhone is connected to another PC, so there's no iPhone's IPv6 address on the PC you runs `start-tunnel`, I guess.

**Comment by Dinip** [comment]
hey @kelvin321, did you ever get your setup to work? 
I was trying a similar setup, but instead of your Mac1 machine, I'm using a linux machine with the ios device connected.

I'm using the --usbmux parameter as mentioned:
`sudo pymobiledevice3 lockdown start-tunnel --usbmux IP:5000`
The `lockdown` tunnel part seems to be working well.

![Image](https://github.com/user-attachments/assets/534def3d-0942-46eb-adc3-5f016e897424)

![Image](https://github.com/user-attachments/assets/8eb2bcef-e8a7-447c-8d4f-7abbd428ae76)

I wonder how this is handled on corellium side, probably something else needs to be "proxied", so the client machine can access the device through the host machine?

**Comment by kelvin321** [comment]
> hey [@kelvin321](https://github.com/kelvin321), did you ever get your setup to work? I was trying a similar setup, but instead of your Mac1 machine, I'm using a linux machine with the ios device connected.
> 
> I'm using the --usbmux parameter as mentioned: `sudo pymobiledevice3 lockdown start-tunnel --usbmux IP:5000` The `lockdown` tunnel part seems to be working well.
> 
> ![Image](https://github.com/user-attachments/assets/534def3d-0942-46eb-adc3-5f016e897424)
> 
> ![Image](https://github.com/user-attachments/assets/8eb2bcef-e8a7-447c-8d4f-7abbd428ae76)
> 
> I wonder how this is handled on corellium side, probably something else needs to be "proxied", so the client machine can access the device through the host machine?

yes, I followed the solution in this thread: https://github.com/doronz88/pymobiledevice3/issues/566

**Comment by doronz88** [comment]
Closing — this usbfluxd multi-Mac setup lacks local IPv6/NCM by design; use `--usbmux` as noted. Not a pymobiledevice3 bug.
